### PR TITLE
Support for relative URLs

### PR DESCRIPTION
**Pull request by tcarrio**
## Description

Provides support for relative URL validation in bueno. This can be used in tandem with the `url` parameter such that specifying a configuration `{ url: true, relativeUrl: true }` would allow a string value that is either a relative URL or an absolute URL.

This is useful for validating URLs used for `fetch` calls, where relative URLs are allowed.

Also provides support for relative URLs in the `platformUrl` and `baseApiUrl` properties for the engine configuration.

## Testing

Included tests for the new `relativeUrl` property in `string-value.test.ts` covering valid/invalid cases as well as valid/invalid cases for the combination of `url` and `relativeUrl`.

### CODE REVIEW

**Comment by ThibodeauJF** [comment]
We've discussed this a bit and think that the urlRegex should support relativeUrls, without requiring a new option.

**Comment by tcarrio** [comment]
Thanks for getting back @ThibodeauJF :wave: 

That would also work, though you lose the convenience of the errors provided upon failure.

In this case, in order to support relative URLs in `headless` it would need to be updated to only use a regex.

I'll make that change accordingly.

File: packages/headless/src/features/configuration/configuration-actions.ts [hunk 1]
```diff
@@ -135,7 +143,11 @@ export const updateAnalyticsConfiguration = createAction(
       enabled: new BooleanValue({default: true}),
       originLevel2: originSchemaOnConfigUpdate(),
       originLevel3: originSchemaOnConfigUpdate(),
-      apiBaseUrl: new StringValue({url: true, emptyAllowed: false}),
+      apiBaseUrl: new StringValue({
+        relativeUrl: true,
+        url: true,
+        emptyAllowed: false,
+      }),
```
Comment (samisayegh) [inline packages/headless/src/features/configuration/configuration-actions.ts:150]: Hi Tom, why should the `platformUrl` and `apiBaseUrl` options support relative values? Could you provide more context behind what is driving this change?

**Comment by tcarrio** [comment]
Hi @ThibodeauJF , would you prefer that `urlRegex` be updated in the underlying validation library (`@coveo/bueno`) then, such that any `StringValue({url: true})` receives this update?

**Comment by ThibodeauJF** [comment]
@tcarrio  Yes that would be my preference, it would make bueno better! Thank you!

For Headless, we can remove the `url:true` like Sami did here: https://github.com/coveo/ui-kit/pull/1042/files#diff-815df8f20ed7874c012c3dde9bf84b718c74ead1eac8b1e6e78cad7d211f52fdL22

It might have been a mistake on our part to enforce url types values here, if customers use a proxy server or some kind or single page web app router, the redirection values might be simpler strings, not urls.

File: packages/headless/src/features/configuration/configuration-actions.ts (hunk 1 above)
Comment (samisayegh) [reply packages/headless/src/features/configuration/configuration-actions.ts:148]: Makes sense, thanks

**Comment by tcarrio** [comment]
Hey team! I'll be picking this up again next week as we have run into a requirement for relative URLs in our project. I'll be updating and submitting this Monday most likely.

**Comment by tcarrio** [comment]
> @tcarrio Yes that would be my preference, it would make bueno better! Thank you!
> 
> For Headless, we can remove the `url:true` like Sami did here: https://github.com/coveo/ui-kit/pull/1042/files#diff-815df8f20ed7874c012c3dde9bf84b718c74ead1eac8b1e6e78cad7d211f52fdL22
> 
> It might have been a mistake on our part to enforce url types values here, if customers use a proxy server or some kind or single page web app router, the redirection values might be simpler strings, not urls.

Following up @ThibodeauJF :wave: I have just pushed my updated to allow relative URLs in bueno and allow the platform and base API URLs to be a non-empty string. I have a quick conflict resolution then this should be good to go!

**Comment by tcarrio** [comment]
Ready for review! :tada:

**Comment by samisayegh** [comment]
Hi Tom, sorry for the back-and-forth on this one. I'm going to close this PR in favour of [this one](https://github.com/coveo/ui-kit/pull/1229) instead. I don't think there is value in headless policing the string, and I can see the regex becoming a maintenance overhead.

Feel free to comment on the other PR. Otherwise, the change should be available on npm shortly after merge.

**Pull request closed without merging**